examples/imagenet/train.py

```diff
@@ -145,7 +145,7 @@ def loss_fn(model):
   step = state.step
   optimizer = state.optimizer
   lr = learning_rate_fn(step)
-  _, (new_model_state, logits), grad = optimizer.compute_gradient(loss_fn)
+  _, (new_model_state, logits), grad = jax.value_and_grad(loss_fn, has_aux=True)(optimizer.target)
```

Comment:
I think this also becomes `(_, (new_model_state, logits), grad)`. 

Can you please run this example (perhaps with tiny batch size), or if you're particular inspired add a test here that runs just one optimization step? (See #176)

Reply:
sorry again. Also, this example won't run because flax.training is not exposed, I changed that too

Reply:
To the extent it's still helpful, ran one epoch of the imagenet example on @joaogui1's branch:

```
I0422 19:10:21.038548 140601894991680 train.py:293] train epoch: 0, loss: 6.7584, accuracy: 0.79
I0422 19:10:43.875979 140601894991680 train.py:315] eval epoch: 0, loss: 6.6295, accuracy: 1.26
```

Reply:
Thanks @danielsuo !